### ISSUE TITLE
OBS Preview blank/crashing, with "Prefer External GPU" enabled, no external display (Mac)

### ISSUE THREAD

**Issue by MisutaaAsriel** [obsproject/obs-studio#4355]
_Edited to include relevant issues ~~and workaround~~ and other information._

## Platform

**Operating System**: macOS Big Sur 11.2.1
**CPU**: Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
**GPU**: Radeon RX 570 4GB (_via_) Sonnet eGFX Breakaway Puck 570S @ 8.0 GT/s, 40 GB/s
**OBS Studio version**: 26.1.2
**OBS Log file**: [obs-log-03162021.txt](https://gist.github.com/MisutaaUrufu/50ef3fe11a716b8945d387f4c5533aa0)
**Apple Crash Reporter Log:** [obs-crash-report-03162021.txt](https://gist.github.com/MisutaaUrufu/87fe7a6d4e8f265bff9411f6be7cb755)

## Expected Behavior
* OBS successfully launches.
* OBS successfully displays the stream preview.

## Current Behavior

* OBS crashes on launch (intermittent).
* OBS launches, but preview area is blank.
* Disabling and re-enabling preview area leads to crash.

## Steps to Reproduce

1. Connect an eGPU unit to the Macintosh system via Thunderbolt 3; **Do not connect external display**.
2. Open OBS's info window in Finder, and enable "Prefer External GPU".
3. Launch OBS.
4. (If successful) Disable, and re-enable, preview.

## Additional Information

~~If one notes in the [crash report], the crashing thread is **CrBrowserMain**, and **CFRunLoop** is involved. This rings a considerable bell, as OBS suffered a similar crashing bug at the launch of Big Sur. Chrome Embedded Framework seems to be the root of the issue... again.~~ Or not. It may just be coincidence, because removing CEF via a framework rename, as was the workaround with previous OBS issues, does not repair the preview window.

Additionally, the OBS log reports an... invalid GL framebuffer operation? It sounds like something is calling an invalid instruction.

Recording/Streaming functionality seems to be unaffected, and if anything, more performant. However, without a preview, editing/creating scenes is not possible. Additionally, it is unknown if browser elements display. — Opening a blank scene does not work however. Nor does disabling hardware acceleration.

## Workaround
~~**Launching OBS with an externally connected display in _Mirror Mode_, which mirrors the contents of the internal display, and disconnecting the display after launch, works. It successfully continues to run until closed without an external display. Issue solely effects OBS when launched without an external display connected.**~~ Update: _This stopped working. Not sure why._

## Screenshot
<img width="1552" alt="image" src="https://user-images.githubusercontent.com/8140068/111379070-b4a36b80-8670-11eb-9388-2adf9c51b544.png">

## Notes
* "Prefer External GPU", when no connected display is found (other than the internal display), allows macOS applications to offload GPU functions to an externally connected GPU, whilst using a display connected to the integrated GPU. In the case of MacBooks, this allows applications to **use the eGPU** unit _**with the internally connected LCD display**_.
* In the case of OBS, this means, rather than rendering the Preview via CPU, and Apple VT choosing which GPU to render with based on performance, OBS and its children will render entirely off the eGFX unit, even when no display is connected to it. This can aid system performance for MacBook users, where eGPU/eGFX usage is most likely to occur.

**Comment by WizardCM** [comment]
Is this the same issue as #3808?

**Comment by MisutaaUrufu** [comment]
> Is this the same issue as #3808?

Possibly, @WizardCM. I didn't notice it when looking, but it looks like their Mac is in "light mode". Does OBS take the system appearance into effect? Their controls in OBS are dark, which made me think that was a separate issue.

These could be separate, could be not.

Also, the workaround I mentioned stopped working. Not sure why.

**Comment by MisutaaUrufu** [comment]
@WizardCM as of macOS version 11.2.3 (20D91), this issue no longer occurs (as of yet). I will ask this be reopened if it comes back, but as of now, it would seem it was a bug with Big Sur, not OBS; I'd recommend that Big Sur uses run 11.2.3 to avoid issues?

**Comment by MisutaaUrufu** [comment]
As of **March 21st, 12:55AM**, the issue **has returned**. I am now puzzled, as I had assumed the update to Big Sur had remedied the issue (it remedied a similar issue with Parallels Desktop's licensing window). — Clearly whatever changed was _temporary_.

Reopening this...

**Comment by MisutaaUrufu** [comment]
Update: **Rebooting** the Mac with **the eGPU connected** _allows OBS to operate_ with an eGPU _and no external display_.

Only after _subsequent disconnects_ and reconnects _of the eGPU_ unit, or when the _eGPU unit is connected **after boot**_, does the issue occur.

**Comment by gxalpha** [comment]
As a note here, no one on the team has the required hardware to reproduce or debug this.
Outside patches are always welcome.

**Comment by gxalpha** [comment]
Actually, as we can't support eGPU's (see the above comment), this issue has no value in staying open.
Again, if someone with this issue is able to fix it, a PR for it would be welcome, but it's not feasible for us to spend the time and effort hunting down a crash we're not even certain that we could fix.
